pinot-query-planner/src/test/java/org/apache/pinot/query/QueryCompilationTest.java

```diff
@@ -392,6 +394,85 @@ public void testDuplicateWithAlias() {
     assertTrue(e.getCause().getMessage().contains("Duplicate alias in WITH: 'tmp'"));
   }
 
+  @Test
+  public void testWindowFunctionsWithCustomWindowFrame() {
+    String queryWithDefaultWindow = "SELECT col1, col2, RANK() OVER (PARTITION BY col1 ORDER BY col2) FROM a";
+    _queryEnvironment.planQuery(queryWithDefaultWindow);
```

Comment:
Is this a complete test for a query ? The expectation is that planning wont throw an exception ? 

Also the same test contains queries that will throw a parse exception ?

Reply:
> Is this a complete test for a query

The queries aren't actually executed, just validated, compiled and optimized. 

> The expectation is that planning wont throw an exception ?

Yes. I can change it to an assertion on `QueryEnvironment.canCompileQuery` to make that more clear perhaps.

> Also the same test contains queries that will throw a parse exception ?

Yes.